yt/loaders.py

```diff
@@ -97,6 +97,21 @@ def load(
     if not fn.startswith("http"):
         fn = str(lookup_on_disk_data(fn))
 
+    # import it here cause it's costly
+    from importlib.metadata import entry_points
+
+    # overly complicated dance to suppress spurious warning
+    # https://github.com/python/importlib_metadata/pull/278
+    entrypoints = entry_points()
+    if hasattr(entrypoints, "select"):
+        external_frontends = entrypoints.select(group="yt.frontends")
+    else:
+        external_frontends = entrypoints.get("yt.frontends", [])
+
+    # Ensure that external frontends are loaded
+    for entrypoint in external_frontends:
+        entrypoint.load()
```

Comment:
could this be done at the top level of the module too ? In case ones has a broken plugin installed I assume it's preferable to error sooner.

Reply:
(on second thought it might create import cycles, so maybe it's not a good idea)

Reply:
((It should _not_ create import cycles, and currently no one is relying on this so we might as well put the responsibility of avoiding cycles on the external frontend if such an anti-pattern is ever needed))

Reply:
moving this to top level makes it even more difficult to test, and the overhead I can measure (~10ms) doesn't seem worth the effort.